### PR TITLE
 feat: support to install fastgpt via helm chart

### LINKED ISSUE #1013: Proposal: support install via helm chart
**例行检查**

[//]: # '方框内填 x 表示打钩'

- [x] 我已确认目前没有类似 features
- [x] 我已确认我已升级到最新版本
- [x] 我已完整查看过项目 README，已确定现有版本无法满足需求
- [x] 我理解并愿意跟进此 features，协助测试和提供反馈
- [x] 我理解并认可上述内容，并理解项目维护者精力有限，**不遵循规则的 features 可能会被无视或直接关闭**

**功能描述**
如题，我希望能通过 helm 的方式来安装，便于在 Kubernetes 中部署。查过 README 等文档没有发现相关描述，如果我有遗漏文档的话，欢迎指出。

与此同时，我计划先写一个 helm chart。维护者如果认为有价值的话，我可以提交到这个仓库。

感谢您的宝贵时间！

**应用场景**
同上。

**相关示例**

### PR DESCRIPTION
**Pull request by LinuxSuRen**
I will do more things once this PR got merged.

* Document
* ~~Currently it cannot support v4.6.6+ due to the MangoDB issues.~~
* ~~Cannot update the `config.json`~~

close #1013 

## For the maintainers
Please add the following to the GitHub secrets before merging this PR:

* `GH_PUBLISH_SECRETS` is the token of below GitHub user
* `GH_USER` is the GitHub user for pushing image

### CODE REVIEW

**Comment by LinuxSuRen** [comment]
Fixed the MangoDB replicaset issue. close https://github.com/labring/FastGPT/issues/1017

**Comment by LinuxSuRen** [comment]
For the [Argo CD]() users, you can use the following to deploy `fastgpt`:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: fastgpt
  namespace: argocd
spec:
  project: default
  source:
    chart: surenpi/fastgpt                            # should be labring/fastgpt once this PR got merged
    repoURL: registry-1.docker.io
    targetRevision: 0.1.0
    helm:
      releaseName: fastgpt
      parameters:
      - name: service.type
        value: 'NodePort'
      - name: service.nodePort
        value: '31449'
      - name: image.pullPolicy
        value: 'Always'
      - name: mongodb.arbiter.enabled
        value: 'false'
      - name: postgresql.image.repository
        value: 'linuxsuren/pgvector'
      - name: postgresql.image.tag
        value: 'master@sha256:ac32e076a4bb8b4f47a4f36f34109f32a58e4de333568756123773e702f52861'
  destination:
    server: "https://kubernetes.default.svc"
    namespace: default
  ignoreDifferences:
  - group: ""
    kind: "ConfigMap"
    name: fastgpt-config
    namespace: default
    jsonPointers:
    - /data
  - group: ""
    kind: "ConfigMap"
    name: fastgpt-env
    namespace: default
    jsonPointers:
    - /data
  syncPolicy:
    syncOptions:
    - CreateNamespace=true
    - RespectIgnoreDifferences=true
    automated:
      prune: true
      selfHeal: true
```

File: files/helm/fastgpt/templates/configmap-env.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+data:
+  DEFAULT_ROOT_PSW: "1234"
+  OPENAI_BASE_URL: "https://api.openai.com/v1"
+  CHAT_API_KEY: "sk-xxxx"
+  DB_MAX_LINK: "5"
+  TOKEN_KEY: "any"
+  ROOT_KEY: "root_key"
+  FILE_TOKEN_KEY: "filetoken"
+  MONGODB_URI: "mongodb://root:123456@{{ include "fastgpt.fullname" . }}-mongodb-headless:27017/fastgpt?authSource=admin"
```
Comment (zzjin) [inline files/helm/fastgpt/templates/configmap-env.yaml:10]: Replace hard coded username&password with `.Values.mongodb.`?

Comment (LinuxSuRen) [reply files/helm/fastgpt/templates/configmap-env.yaml:10]: Good catch!

File: files/helm/fastgpt/templates/configmap-env.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
```
Comment (lingdie) [inline files/helm/fastgpt/templates/configmap-env.yaml:1]: Please use secrets to store those values.

Comment (LinuxSuRen) [reply files/helm/fastgpt/templates/configmap-env.yaml]: Sure.

image

Comment (LinuxSuRen) [reply files/helm/fastgpt/templates/configmap-env.yaml]: See also https://github.com/labring/FastGPT/pull/1023/commits/dea1a3460e98ce4f85e49b5a12e5cc977fbd533c

**Review by lingdie — APPROVED** [review]

**Comment by zzjin** [comment]
@LinuxSuRen We already have github secrets:
Use `github.repository_owner` for `GH_USER` and `secrets.GH_PAT` for `GH_PUBLISH_SECRETS`.
Thanks.

File: .github/workflows/helm-release.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          echo ${{ secrets.GH_PUBLISH_SECRETS }} | helm registry login ghcr.io -u ${{ secrets.GH_USER }} --password-stdin
+          export APP_VERSION=${{ steps.vars.outputs.tag }}
+          export HELM_VERSION=${{ steps.vars.outputs.tag }}
+          if [[ ! "$line" =~ ^v ]]
+          then
+            unset APP_VERSION
+            unset HELM_VERSION
+          fi
+          helm package files/helm/fastgpt --version ${HELM_VERSION}-helm --app-version ${APP_VERSION} -d bin
+          helm push bin/fastgpt-${HELM_VERSION}-helm.tgz oci://${HELM_REPO}
```
Comment (zzjin) [inline .github/workflows/helm-release.yaml:33]: `HELM_REPO` env var does not exists?

Comment (LinuxSuRen) [reply .github/workflows/helm-release.yaml:33]: Sorry about that.

File: .github/workflows/helm-release.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+name: Release
+
+on:
+  push:
+    tags:
```
Comment (zzjin) [inline .github/workflows/helm-release.yaml:5]: Should we build helm charts only at specific tag?

like:
`    tags:
      - 'v*.*.*'
`
@c121914yu

File: files/helm/fastgpt/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+apiVersion: v2
+name: fastgpt
```
Comment (zzjin) [inline files/helm/fastgpt/Chart.yaml:2]: Maybe it will force replace fastgpt‘s docker image?
shaul we change to `fastgpt-helm`?

Comment (LinuxSuRen) [reply files/helm/fastgpt/Chart.yaml:2]: The image tags are different. The helm chart image tag has the suffix `-helm`.

See also https://github.com/labring/FastGPT/pull/1023/files#diff-e55e8b28bf787ec17d6f2b235a396d3256da0ec8b9818603c61b46197d84e6c0R33

File: .github/workflows/helm-release.yaml (hunk 4 above)
Comment (zzjin) [reply .github/workflows/helm-release.yaml:5]: After discussion with @c121914yu , we should change matched tags here @LinuxSuRen

Comment (LinuxSuRen) [reply .github/workflows/helm-release.yaml:5]: No problem from me. So you perfer to use `'v*.*.*`?

Comment (zzjin) [reply .github/workflows/helm-release.yaml:5]: To match: https://github.com/labring/FastGPT/blob/main/.github/workflows/fastgpt-image.yml#L8 ?

Comment (LinuxSuRen) [reply .github/workflows/helm-release.yaml:5]: Do you recommend to add the file path filter or not? https://github.com/labring/FastGPT/blob/911512b36d7353ebe6cc16428976210dcd8477d3/.github/workflows/fastgpt-image.yml#L5

Such as:

```yaml
on:
  workflow_dispatch:
  push:
    paths:
      - 'files/helm/fastgpt/**'
    tags:
      - 'v*.*.*'
```

Comment (zzjin) [reply .github/workflows/helm-release.yaml:5]: Seems just cannot match these paths, so keep `*` just works.

**Review by zzjin — APPROVED** [review]

**Comment by c121914yu** [comment]
image
You might want to look at the action in this preview-images

**Comment by LinuxSuRen** [comment]
> You might want to look at the action in this preview-images

Got it.

**Pull request merged**